autogpt/config/ai_config.py

```diff
@@ -130,6 +130,14 @@ def construct_full_prompt(
                 continue
             prompt_generator = plugin.post_prompt(prompt_generator)
 
+        # If execute_local_commands enabled, add OS info to prompt
+        if cfg.execute_local_commands:
+            os_name = platform.system()
+            if os_name == "Darwin":
+                os_name = "MacOS"
+            os_version = platform.release()
```

Comment:
What does `release()` give on MacOS and for Linux distros?

Reply:
macOS:

```python
import platform

print(platform.system())  # Output: 'Darwin'
print(platform.release())  # Output: '20.3.0' (This is the kernel version)
```

Linux (e.g., Ubuntu):

```python
import platform

print(platform.system())  # Output: 'Linux'
print(platform.release())  # Output: '5.4.0-65-generic' (This is the kernel version)
```

I think that having the kernel version is not the best way. I found that we can use something like the following:
```python
import platform
import sys

def get_platform_info():
    system = platform.system()
    version = platform.version()

    if system == "Linux":
        import distro
        distro_info = distro.name(pretty=True)
        version = f"{distro_info} ({version})"
    elif system == "Darwin":
        system = "macOS"
        version = platform.mac_ver()[0]

    return system, version

platform_name, platform_version = get_platform_info()
print(f"Platform Name: {platform_name}")
print(f"Platform Version: {platform_version}")
```

That function will return something similar to the following:

Windows:

```
Platform Name: Windows
Platform Version: 10.0.19041
```

macOS:

```
Platform Name: macOS
Platform Version: 11.2.3
```

Linux (e.g., Ubuntu):

```
Platform Name: Linux
Platform Version: Ubuntu 20.04.2 LTS (5.4.0-65-generic)
```

TO be honest I prefer this last option, I can try integrating something similar after execute_local_commands is checked.
Let me know what do you think it's best.

Reply:
Nice comparison! And how do they compare to `platform.platform()`? (with/without args)

Reply:
After reading some documentation found that they return the following:
(Couldn't test them all right now, just confirmed the windows one and the example below looks correct)

Windows:

```python
import platform

print(platform.platform())  # Output: 'Windows-10-10.0.19045-SP0'
print(platform.platform(terse=True))  # Output: 'Windows-10'
```

macOS:

```python
import platform

print(platform.platform())  # Output: 'Darwin-20.3.0-x86_64-i386-64bit'
print(platform.platform(terse=True))  # Output: 'Darwin'
```

Linux (e.g., Ubuntu):

```python
import platform

print(platform.platform())  # Output: 'Linux-5.8.0-50-generic-x86_64-with-glibc2.32'
print(platform.platform(terse=True))  # Output: 'Linux'
```

Reply:
The macOS one is not correct. Someone in #testing on Discord was kind enough to run this for me:
```py
print(platform.platform(True))
print(platform.platform(terse=True))
```

Output:
```
macOS-10.15.7-x86_64-i386-64bit
macOS-10.15.7
```

Reply:
Can confirm on my mac I get 
```
print(platform.platform())  # macOS-12.6.3-arm64-arm-64bit
print(platform.platform(terse=True)) # macOS-12.6.3
print(platform.system())  # Darwin
```